pkg/bloomgateway/querier.go

```diff
@@ -119,30 +119,28 @@ func (bq *BloomQuerier) FilterChunkRefs(ctx context.Context, tenant string, from
 	preFilterSeries := len(grouped)
 
 	// Do not attempt to filter chunks for which there are no blooms
-	if bq.cfg.MinTableOffset > 0 {
-		minAge := truncateDay(model.Now()).Add(-1 * config.ObjectStorageIndexRequiredPeriod * time.Duration(bq.cfg.MinTableOffset-1))
-		if through.After(minAge) {
-			level.Debug(logger).Log(
-				"msg", "skip too recent chunks",
-				"tenant", tenant,
-				"from", from.Time(),
-				"through", through.Time(),
-				"responses", 0,
-				"preFilterChunks", preFilterChunks,
-				"postFilterChunks", preFilterChunks,
-				"filteredChunks", 0,
-				"preFilterSeries", preFilterSeries,
-				"postFilterSeries", preFilterSeries,
-				"filteredSeries", 0,
-			)
-
-			bq.metrics.chunksTotal.Add(float64(preFilterChunks))
-			bq.metrics.chunksFiltered.Add(0)
-			bq.metrics.seriesTotal.Add(float64(preFilterSeries))
-			bq.metrics.seriesFiltered.Add(0)
-
-			return chunkRefs, false, nil
-		}
+	minAge := model.Now().Add(-1 * (config.ObjectStorageIndexRequiredPeriod*time.Duration(bq.cfg.BuildTableOffset) + bq.cfg.BuildInterval))
```

Comment:
Just going through before/after here for my own knowledge, assuming that the build interval is 8h:

| MinTableOffset | minAge Before      | minAge After | 
| -------------- | ------------------ | ------------ | 
| 0              | N/A                | `now - 8h`   |
| 1              | Midnight today     | `now - 32h`  | 
| 2              | Midnight yesterday | `now - 56h`  | 

This definitely seems better, though we might want to do `2*bq.cfg.BuildInterval` to make sure we never query for blocks that the planner hasn't gotten to yet. I'm not sure if it's necessary, though, so it's up to you if you think it makes sense.

Reply:
Addressed with b9aedbf11ff2df249a6c0b8a5c1edfaed3bc2eb2